plugins/modules/system/puppet.py

```diff
@@ -176,7 +176,9 @@ def main():
                                                                 'stdout',
                                                                 'syslog']),
             # internal code to work with --diff, do not use
-            show_diff=dict(type='bool', default=False, aliases=['show-diff']),
+            show_diff=dict(
```

Comment:
We should probably deprecate the whole option, and use `module._diff` instead of this option.

Reply:
done, commit coming up soon

Reply:
Shouldn't we also add the explanation to the changelog that it'll be returned via the standard `diff` mechanism.
Anyway, to see this PR merged ASAP, if i were russoz, i'd do this in another "dedicated" PR. What do you think?

Reply:
It won't be returned by the standard `diff` mechanism, because the module does not support that.